### ISSUE TITLE
MongoDB Plugin - Query Information Enhancement

### ISSUE THREAD

**Issue by dwinrick-lever** [DataDog/dd-trace-js#1592]
With the MongoDB plugin, as it is today, I get events like this for each query:

`{"$or":"?","a":"?","b":"?","c":"?","d":"?","e":"?"}`

What would be cool is an option to get something like this:

`{"$or":"object","a":"null","b":"boolean","c":"object","d":"boolean","e":"string"}`

This would better help us take action on the reporting from DataDog to design our indexes.

I'd imagine this would be configurable to not break existing customers' reports.

This is a pretty straight forward change - if the DataDog team is up for it I'm happy to put together a PR.

EDIT - When I say "boolean" and "string" above I literally mean it to say "boolean" or "string" in the event, not the actual value.

**Comment by rochdev** [comment]
Not necessarily. A pattern we've been using in other places is to compute and set a tag only when it wasn't already set in a hook, so that could be an option. If the use case is common enough though, it could still warrant its own configuration option.

**Comment by BridgeAR** [comment]
This is now implemented and shipped in our latest tracer. Closing as resolved. I hope the feature works well for you :)
